### ISSUE TITLE
Plex categories don't show up anymore in home menu

### ISSUE THREAD

**Issue by gdachs** [marcelveldt/skin.titan#126]
Sorry if I tell nonsense, I am no python programmer or xbmc skinner and no native speaker. 

I believe the problem is starting in this line:
`1080i/Home.xml:	<onload condition="System.HasAddon(plugin.video.plexbmc)">XBMC.RunScript(script.titanskin.helpers,UPDATEPLEXLINKS)</onload>`
When you moved the function updatePlexLinks() from the MainModule to HomeMonitor the code for UPDATEPLEXLINKS vanished from default.py so the onload call does nothing and HomeMonitor calls updatePlexLinks only after 15 Minutes.

Regards
Gerald

**Comment by marcelveldt** [comment]
Well, the updatePlexLinks method is called directly on service startup. Your are right it runs every 15 minutes but it also runs directly on first run, because of the initial count set to 899.
What version of the skin are you running ?

**Comment by gdachs** [comment]
So the <onload> line can be removed?

I will look when I am at home which version I use. I use isengard and got the skin from the official repository.

Currently I have the issue that the plex links don't show up on boot. I have to switch to the skin amber first, that shows the links, and then switch back to titan to get the links there.

**Comment by marcelveldt** [comment]
It is already removed in the new skin version. The version on official Kodi repo is a little outdated. I have already submitted the new version to the Kodi devs but because of the holidays it takes a little longer to update.

You may want to give the beta version of the skin a try, see the first post on the Titan forum thread:
http://forum.kodi.tv/showthread.php?tid=201061

Your problem could be a timing issue indeed. It's working fine on my testsetup but that only has a few collections just for testing and the Plex server is on my developmachine so very low latency.

If you still have the issue with the beta version you might want to try to lower the initial count a bit.
Or: In the method updatePlexLinks (around line 103 of HomeMonitor.py) you will see this line:
while (count < 80 and self.win.getProperty("plexbmc.0.title") == ""):

Maybe increase that one to 120 or so. It waits fox max 20 seconds for the plexplgin to respond with the correct values. Maybe that is not enough?

**Comment by gdachs** [comment]
I tried now Titan beta. Here it even doesn't help to start amber before. No Plex tiles are shown.

The problem is the line 105 in HomeMonitor.py. The condition is never true. It is the part Skin.HasSetting(SmartShortcuts.plex). I couldn't fine out how to set this in the Skin settings.
 How do I do that? 

I have tried it without this conditions and the Plex tiles are back :)

But, the "on deck" and "recently added" things don't show up. 

Maybe we could chat about this? I am in #kodi mostly in the evening

**Comment by marcelveldt** [comment]
Sorry, I don't have chat :-)
So, we must figure it out this way I'm afraid.

In the skin you must enable "smart shortcuts" for Plex. After that, add the plex tiles from the available shortcuts.

The main issue here is that the skin's helper service got updated on the repo before the skin was. It seems to be that the Kodi dev that adds the skins to the repo is on holiday or something.

**Comment by gdachs** [comment]
Chatzilla for Firefox works on every platform ;), but anyway ...

My problem is, that I don't find that. In advanced settings are smart shortcuts for playlists and favorites, but I can't find them for plex. Tell me in which menu to look.

**Comment by marcelveldt** [comment]
Are you sure you switched to the beta version of the skin ?
The beta version contains that setting. Like I said it's a backwards compatibility thing. The helper service is updated on the repo while the skin isn't....

If you want I can build you a zip for the stable version of 3.4.0

**Comment by marcelveldt** [comment]
Here is a link to the zip of the 3.4.0 stable version...

https://github.com/marcelveldt/xbmc.skin.titan/raw/master/skin.titan.zip

**Comment by gdachs** [comment]
Upps, sorry, was really back on the non beta. Switched now to the beta, the option is set but the tiles don't show up, have to debug again.

I will install your version 3.4.0 first. Should I clean up userdata?

**Comment by marcelveldt** [comment]
No need to clean up userdata *I think*...
Please install my zipversion, that way you will have all of your previous settings including the tiles.

The beta has a seperate ID so it has it's own settings.

**Comment by gdachs** [comment]
Tried the 3.4.0. It has the smart shortcuts setting for plex and it is active by default, but the Tiles don't show up. Have to debug again, maybe I have tomorrow more.

**Comment by marcelveldt** [comment]
Go to skin settings --> configure menu shortcuts
Press Add 
You should see "Smart shortcuts for Plexbmc" listed there. 
Can you add them to your shortcuts ?

I don't understand why it didn't migrate your previous settings... Did you reset the shortcuts or skin settings by any chance ?

**Comment by gdachs** [comment]
I have never done anything to the skin settings before, the tiles was always there without any user interaction. In Amber I had to use "Go Plex". I really have to do something in the settings to get the tiles? Why did it work before without?

It seems I have misunderstood this completely. When I add a tile by hand, then it works. Sorry for the noise.

**Comment by marcelveldt** [comment]
That's because I changed the whole "plex mode" to use the more universal "smart shortcuts" system I designed which will work with all addons. Before I had 10.000's of lines of code for the "plex mode", "mediabrowser mode" and stuff like that. It's now all integrated in a more flexible approach. This saves a lot of "hacky" code in the skin, giving the skin more stability and speed.

That said, while migrating the whole stuff to the new approach I had in mind that users who prevously used the old approach should be automatically merged so they won't be lost. As it turns out I was wrong. So thanks to your issue report I now know that I should improve this to keep things working for existing users.

Is everything working OK for you now after you added your Plex tiles back to the menu ?
